### PR TITLE
Downloading testcases and archiving fuzzjob now in background

### PR DESCRIPTION
**Pull request by p4cx**
Web user interface now is usable and not blocked, if you download a big testcase set or archiving an old fuzzjob.

### CODE REVIEW

**Review by TomSie — APPROVED** [review]

File: srv/fluffi/data/fluffiweb/app/controllers.py [hunk 1]
```diff
@@ -138,6 +139,90 @@ def getDownloadPath():
...
+class CreateTestcaseArchive(threading.Thread):
+    def __init__(self, projId, name_list, count_statement_list, statement_list):
+        super().__init__()
+        self.progress = 0
+        self.projId = projId
+        self.name_list = name_list
+        self.count_statement_list = count_statement_list
+        self.statement_list = statement_list
+        self.max_val_list = []
+        # status: 0 = default/running, 1 = success, 2 = error
```
Comment (p0wer0xff) [inline srv/fluffi/data/fluffiweb/app/controllers.py:151]: we could make this more readable by using an [Enum](https://www.python.org/dev/peps/pep-0435/)
`from enum import Enum; Status = Enum('Status', 'running success error')`

File: srv/fluffi/data/fluffiweb/app/controllers.py [hunk 2]
```diff
@@ -1185,3 +1270,63 @@ def deleteDatabase(fuzzjobName):
     finally:
         connection.close()
         engine.dispose()
+
+
+class ArchiveProject(threading.Thread):
+    def __init__(self, projId):
+        super().__init__()
+        self.projId = projId
+        # status: 0 = default/running, 1 = success, 2 = error, 3 = finish
```
Comment (p0wer0xff) [inline srv/fluffi/data/fluffiweb/app/controllers.py:1279]: see above

File: srv/fluffi/data/fluffiweb/app/__init__.py [hunk 3]
```diff
@@ -34,7 +34,7 @@
 
 scheduler = BackgroundScheduler()
 scheduler.configure(timezone = utc)
-scheduler.add_job(deleteZipAndTempFiles, 'interval', minutes=3)
+#scheduler.add_job(deleteZipAndTempFiles, 'interval', minutes=3)
```
Comment (p0wer0xff) [inline srv/fluffi/data/fluffiweb/app/__init__.py:37]: pls check if this is on purpose

**Review by p0wer0xff — APPROVED** [review]
with the discussed drawback of using a single file, I think we can merge this.

**Pull request merged**